### ISSUE TITLE
get_num_terms function run forever if TDC term not found.

### ISSUE THREAD

**Issue by cyrushu** [Valdes-Tresanco-MS/gmx_MMPBSA#98]
The function is located at:
https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/blob/master/GMXMMPBSA/amber_outputs.py#L1471

**Describe the issue**
when TDC term is not found accidentally, the function never break in the while loop

**Suggested modification**
This function loads all the files into memory and processes the TDC count. Happy to start a pr if needed.
```python
    def get_num_terms(self):
        """ Gets the number of terms in the output file """
        with open('%s.%d' % (self.basename, 0), 'r') as decfile:
            lines = decfile.readlines()
        num_terms = 0
        FLAG = False
        for line in lines:
            if line[0:3] == 'TDC':
                num_terms += 1
                FLAG = True
            else:
                if FLAG:
                    break
            # We've now gotten to the end of the Total Decomp Contribution,
            # so we know how many terms we have
        if not FLAG:
            raise NotImplementedError("{}.{} have 0 TDC starts".format(self.basename, 0))
        return num_terms
```

**Comment by Valdes-Tresanco-MS** [comment]
Thanks for reporting this problem.
It has never happened to me ... Could you put me in context with an example?

Cheers!
Mario S.

**Comment by cyrushu** [comment]
> Thanks for reporting this problem. It has never happened to me ... Could you put me in context with an example?
> 
> Cheers! Mario S.

Sorry for the late response. I have not preserved the context. 

When the Ligand is located far from the pocket/protein, There would be no residue around the ligand. If "TDC" does not exist in the temp file, the problem occurs. 

Best wishes,
Yilang.

**Comment by cyrushu** [comment]
> 

Could Simply delete all the TDC in the temp file and that problem would happen.

**Comment by Valdes-Tresanco-MS** [comment]
In the next few days I will try to review this problem and finalize version 1.5.0.
In such case, I will let you know for a PR.
Best!
Mario S.

**Comment by Valdes-Tresanco-MS** [comment]
Available as of v1.5.0
